spring-thrift-starter/src/main/java/info/developerblog/spring/thrift/client/pool/ThriftClientPooledObjectFactory.java

```diff
@@ -41,20 +47,40 @@ public TServiceClient create(ThriftClientKey key) throws Exception {
 
         TProtocol protocol;
 
-        if (Strings.isNullOrEmpty(endpoint)) {
+        Span span = this.tracer.createSpan(key.getServiceName());
+        Map<String, String> headers = new HashMap<>();
+
+        if (span == null) {
```

Comment:
@marcingrzejszczak in FeignRequestTemplateInjector this could be, so we do the same logic